frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -997,6 +999,14 @@ private void removeContainerRequest(TaskStatus taskStatus) {
     } catch (Exception e) {
       LOGGER.logError(e, "%s: Failed to removeContainerRequest", taskLocator);
     }
+    if(request.getNodes() != null && request.getNodes().size() != 0) {
+      try {
+        ResourceDescriptor resourceDescriptor = ResourceDescriptor.fromResource(request.getCapability());
+        gpuAllocationManager.removeContainerRequest(resourceDescriptor, request.getNodes());
+      } catch (Exception e) {
+        LOGGER.logError(e, "%s: Failed to gpuAllocationManager.removeContainerRequest", taskLocator);
+      }
+    }
```

Comment:
try {
      rmClient.removeContainerRequest(request);
    } catch (Exception e) {
      LOGGER.logError(e, "%s: Failed to rmClient.removeContainerRequest", taskLocator);
    }

    try {
      gpuAllocationManager.removeContainerRequest(request);
    } catch (Exception e) {
      LOGGER.logError(e, "%s: Failed to gpuAllocationManager.removeContainerRequest", taskLocator);
    }

Reply:
What do you mean ?

---
In reply to: [158595991](https://github.com/Microsoft/pai/pull/128#discussion_r158595991) [](ancestors = 158595991)